### ISSUE TITLE
Updating relationship: "Schema is not registered for a resource at path"

### ISSUE THREAD

**Issue by oskar1233** [FriendsOfCake/crud-json-api#23]
# The problem

When I'm trying to update my resource's `belongsTo` association I'm running into `500` error with `Schema is not registered for a resource at path 'thing'.` message.

# Kinda workaround

However, when I remove `src/Listener/JsonApiListener.php` line 373 (`JsonApiListener::_insertBelongsToDataIntoEventFindResult`)
```
$key = Inflector::singularize($key);
```
...everything goes well, except that in response association gets rendered in attributes and in plural like so:
```
"attributes": {
    "email": "test",
    "password": "",
    "things": {
        "id": 1
    }
}
```
I'm using v0.2.1 which I believe is the newest.

**Comment by bravo-kernel** [comment]
Mostly that error message is related to an Entity or Table missing for one of the associated models.

Could you verify they all exist?

**Comment by oskar1233** [comment]
They do. I forgot to mention that I tested fetching the data directly. 

I think that problem is with singularization and pluralization. My table name's plural while error says that singular named schema doesn't exist - which is indeed true.

**Comment by bravo-kernel** [comment]
Aha, missed the updating part ;) Could you paste the full json body data you are POSTing?

**Comment by oskar1233** [comment]
I'm PATCHing it to be clear ;). Sure, here it is:
```
{
	"data": {
		"type": "user",
		"id": "1",
		"relationships": {
			"client": {
				"data": {"type": "client", "id": "1"}
			}
		}
	}
}
```

And that's what I receive (only few first traces):
```
{
    "errors": [
        {
            "code": "500",
            "title": "Internal Server Error",
            "detail": "Schema is not registered for a resource at path 'client'."
        }
    ],"debug": {
        "class": "InvalidArgumentException",
        "trace": [
            {
                "file": "/app/vendor/neomerx/json-api/src/Encoder/Parser/Parser.php",
                "line": 163,
                "function": "getSchema",
                "class": "Neomerx\\JsonApi\\Encoder\\Parser\\Parser",
                "type": "->",
                "args": [
                    {
                        "id": 1
                    },
                    []
                ]
            },
            {
                "file": "/app/vendor/neomerx/json-api/src/Encoder/Parser/Parser.php",
                "line": 185,
                "function": "parseData",
                "class": "Neomerx\\JsonApi\\Encoder\\Parser\\Parser",
                "type": "->",
                "args": []
            },
            {
                "file": "/app/vendor/neomerx/json-api/src/Encoder/Parser/Parser.php",
                "line": 135,
                "function": "parseData",
                "class": "Neomerx\\JsonApi\\Encoder\\Parser\\Parser",
                "type": "->",
                "args": []
            },
            {
                "file": "/app/vendor/neomerx/json-api/src/Encoder/Encoder.php",
                "line": 226,
                "function": "parse",
                "class": "Neomerx\\JsonApi\\Encoder\\Parser\\Parser",
                "type": "->",
                "args": [
                    {
                        "email": "test",
                        "password": "",
                        "client_id": 1,
                        "client": {
                            "id": 1
                        }
                    }
                ]
            },
...
```

**Comment by birdy247** [comment]
I am facing this same issue. @oskar1233 did you find any solution?  I also have a feeling it is to do with the singular v plural

**Comment by oskar1233** [comment]
@birdy247 Not really; I haven't dig into this more as my record gets updated. I perform second query to fetch the data if needed.

It can be `neomerx/json-api` issue as well; I assumed the problem is with `crud-json-api` without a reason (okay, maybe because of popularity/maturity of the libs).

**Comment by oskar1233** [comment]
@birdy247 if you have a free moment you can check if adding conditional singularizing instead of permanent one in mentioned code fragment will do the job.

**Comment by birdy247** [comment]
@oskar1233 my branch fixes this and is hopefully going to be merged

**Comment by oskar1233** [comment]
Great, thank you for fixing and mention!

**Comment by bravo-kernel** [comment]
Resolved in #25, available in release 0.2.2
